hi-ml-histopathology/testhisto/testhisto/utils/test_plots_utils.py

```diff
@@ -88,11 +89,43 @@ def test_plots_handler_plots_only_desired_plot_options(plot_options: Collection[
 
 
 def test_save_conf_matrix_integration(tmp_path: Path) -> None:
+    matplotlib_logger = logging.getLogger('matplotlib')
+    matplotlib_logger.setLevel(logging.WARNING)
     results = {
         ResultsKey.TRUE_LABEL: [0, 1, 0, 1, 0, 1],
         ResultsKey.PRED_LABEL: [0, 1, 0, 0, 0, 1]
     }
     class_names = ["foo", "bar"]
+
     save_confusion_matrix(results, class_names, tmp_path)
     file = Path(tmp_path) / "normalized_confusion_matrix.png"
     assert file.exists()
+
+    # check that an error is raised if true or predicted labels include indices greater than the expected number of
+    # classes
+    invalid_results_1 = {
+        ResultsKey.TRUE_LABEL: [0, 1, 0, 1, 0, 2],
+        ResultsKey.PRED_LABEL: [0, 1, 0, 0, 0, 1]
```

Comment:
I am not sure this is actually an error. The set of true label might be {0, 1, 2} but the model could have predicted false labels that are only in {0, 1}, I wouldn't consider that an error especially when we use a limited number of val/test batches that might not include all possible labels...

Reply:
The error here is that the code will try to index class_names at index 2, which is out of range